futurenhs.api/FutureNHS.Api/Services/Admin/AdminGroupService.cs

```diff
@@ -0,0 +1,322 @@
+            IGroupImageService imageService,
+            IGroupCommand groupCommand,
+            IHtmlSanitizer htmlSanitizer)
+        {
+            _systemClock = systemClock ?? throw new ArgumentNullException(nameof(systemClock));
+            _blobStorageProvider = blobStorageProvider ?? throw new ArgumentNullException(nameof(blobStorageProvider));
+            _permissionsService = permissionsService ?? throw new ArgumentNullException(nameof(permissionsService));
+            _fileTypeValidator = fileTypeValidator ?? throw new ArgumentNullException(nameof(fileTypeValidator));
+            _groupCommand = groupCommand ?? throw new ArgumentNullException(nameof(groupCommand));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _imageService = imageService ?? throw new ArgumentNullException(nameof(imageService));
+            _htmlSanitizer = htmlSanitizer ?? throw new ArgumentNullException(nameof(htmlSanitizer));
+        }
+
+        public async Task CreateGroupAsync(Guid userId, Stream requestBody, string? contentType, CancellationToken cancellationToken)
+        {
+            var userCanPerformAction = await _permissionsService.UserCanPerformActionAsync(userId, AddGroupRole, cancellationToken);
+            if (userCanPerformAction is not true)
+            {
+                _logger.LogError($"Error: CreateGroupAsync - User:{0} does not have access to add group:{1}", userId);
+                throw new SecurityException($"Error: User does not have access");
+            }
+
+            var (group, image) = await AdminUploadGroupImageMultipartContent(userId, requestBody, contentType, cancellationToken);
+ 
+            if (image is not null)
+            {
+                var imageValidator = new ImageValidator();
+                var imageValidationResult = await imageValidator.ValidateAsync(image, cancellationToken);
+                if (imageValidationResult.Errors.Count > 0)
+                    throw new ValidationException(imageValidationResult);
+                try
+                {
+                    var imageId = await _imageService.CreateImageAsync(image);
+                    group = group with { ImageId = imageId };
+                }
+                catch (DBConcurrencyException ex)
+                {
+                    _logger.LogError(ex, $"Error: CreateImageAsync - Error creating image {0}", null);
+                    if (image is not null)
+                    {
+                        await _blobStorageProvider.DeleteFileAsync(image.FileName);
+                        await _imageService.DeleteImageAsync(image.Id);
+                    }
+                    throw new DBConcurrencyException("Error: User request to create an image was not successful");
+                }
+            }
+
+            var groupValidator = new GroupValidator();
+            var groupValidationResult = await groupValidator.ValidateAsync(group, cancellationToken);
```

Comment:
Slug is being passed into the Api but theres no check to ensure its unique and the DB doesn't have a unique key constraint on Slug.

Reply:
Ignore this, the unique constraint is set on the index